autoreject/autoreject.py

```diff
@@ -775,198 +824,274 @@ def fit(self, epochs):
+        """
+        return self.fit(epochs).transform(epochs, return_log=return_log)
+
+
+def _check_fit(epochs, threshes_, picks_):
+    if not all(epochs.ch_names[pp] in threshes_
+       for pp in picks_):
+            raise ValueError(
+                'You are passing channels which were not present '
+                'at fit-time. Please fit it again, this time '
+                'correctly.')
+
+
+def _apply_interp(reject_log, epochs, threshes_, picks_,
+                  verbose):
+    _check_fit(epochs, threshes_, picks_)
+    interp_channels = _get_interp_chs(
+        reject_log.labels, reject_log.ch_names, picks_)
+    _interpolate_bad_epochs(
+        epochs, interp_channels=interp_channels,
+        picks=picks_, verbose=verbose)
+
+
+def _apply_drop(reject_log, epochs, threshes_, picks_,
+                verbose):
+    _check_fit(epochs, threshes_, picks_)
+    if np.any(reject_log.bad_epochs):
+        epochs.drop(np.nonzero(reject_log.bad_epochs)[0],
+                    reason='AUTOREJECT')
+    else:
+        warnings.warn(
+            "No bad epochs were found for your data. Returning "
+            "a copy of the data you wanted to clean. Interpolation "
+            "may have been done.")
+
+
+def _get_interp_chs(labels, ch_names, picks):
+    """Convert labels to channel names.
+    It returns a list of length n_epochs. Each entry contains
+    the names of the channels to interpolate.
+
+    labels is of shape n_epochs x n_channels
+    and picks is the sublist of channels to consider.
+    """
+    interp_channels = list()
+    assert labels.shape[1] == len(ch_names)
+    assert labels.shape[1] > np.max(picks)
+    idx_nan_in_row = np.where(np.any(~np.isnan(labels), axis=0))[0]
+    np.testing.assert_array_equal(picks, idx_nan_in_row)
+    for this_labels in labels:
+        interp_idx = np.where(this_labels == 2)[0]
+        interp_channels.append([ch_names[ii] for ii in interp_idx])
+    return interp_channels
+
+
+class RejectLog(object):
+    """The Rejection Log.
+
+    Parameters
+    ----------
+    bad_epochs : array-like, shape (n_epochs,)
+        The boolean array with entries True for epochs that
+        are marked as bad.
+    labels : array, shape (n_channels, n_epochs)
+        It contains integers that encode if a channel in a given
+        epoch is good (value 0), bad (1), or bad and interpolated (2).
+    ch_names : list of str
+        The list of channels corresponding to the rows of the labels.
+    """
+
+    def __init__(self, bad_epochs, labels, ch_names):
+        self.bad_epochs = bad_epochs
+        self.labels = labels
+        self.ch_names = ch_names
+        assert len(bad_epochs) == labels.shape[0]
+        assert len(ch_names) == labels.shape[1]
+
+    def plot(self):
+        """Plot."""
+        import matplotlib.pyplot as plt
+
+        plt.figure(figsize=(12, 6))
+        plt.imshow(self.labels, cmap='Reds',
+                   interpolation='nearest')
+        # XXX to be fixed
```

Comment:
ah ... this is still broken. I think this is a small blocker before we can merge this PR. Otherwise, everything else looks in pretty good shape

Reply:
never mind ... I spoke too soon. I like the viz now!